### PR TITLE
Add SLSA 1.0 attestation support to cosign. Closes #2860

### LINKED ISSUE #2860: Add support for SLSA 1.0 predicate
Cosign supports the v0.2 SLSA predicate through the `--type slsaprovenance` flag on the `attest` and `attest-blob` commands. SLSA 1.0 Release Candidate 1 (RC1) is out and RC2 should be coming out today. By the end of this month (April 2023) it is expected that SLSA v1.0 will be officially released. We don't expect any new major changes to the SLSA provenance predicate.

For reference, here's the specification for the predicate: https://slsa.dev/provenance/v1-rc1

### PR DESCRIPTION
**Pull request by ziel**
#### Summary
This change adds support for the SLSA 1.0 provenance predicate (https://slsa.dev/spec/v1.0/provenance).

`cosign attest` and related commands cannot validate it currently, and the `custom` type must be used for it instead. After this change, it will be possible to specify the new provenance attestation type as `slsaprovenance1`.

Related issue: https://github.com/sigstore/cosign/issues/2860

#### Release Note
Added a new type `slsaprovenance1` to support slsa 1.0 provenance in attest and verify commands.

#### Documentation
The attestation types do not appear to be documented at https://docs.sigstore.dev so there is no related change there.

### CODE REVIEW

**Comment by haydentherapper** [comment]
I think this is a good way to approach this, and we can plan to change the behavior of the flags in a later release as a breaking change, where `slsaprovenance` will generate a v1.0 attestation and we'll drop support for v0.2.

cc @znewman01

File: pkg/cosign/attestation/attestation.go [hunk 1]
```diff
@@ -99,7 +100,7 @@ type GenerateOpts struct {
 }
 
 // GenerateStatement returns an in-toto statement based on the provided
-// predicate type (custom|slsaprovenance|spdx|spdxjson|cyclonedx|link).
+// predicate type (custom|slsaprovenance|slsaprovenance1.0|spdx|spdxjson|cyclonedx|link).
```
Comment (znewman01) [inline pkg/cosign/attestation/attestation.go:103]: nit: `slsaprovenance1`.

**Review by znewman01 — CHANGES_REQUESTED** [review]
Maybe to ease the transition can we add a `slsaprovenance0.2` option that's just an alias for the current behavior?

Then we could at some future point switch over `slsaprovenance` to point to `slsaprovenance1`.

Otherwise looks great, thanks for the change!

File: pkg/cosign/attestation/attestation.go (hunk 1 above)
Comment (ziel) [reply pkg/cosign/attestation/attestation.go:103]: oh oops! will fix

**Review by haydentherapper — APPROVED** [review]

**Comment by ziel** [comment]
hi @znewman01 -- i added a patch related to your change request, if you have a moment to peek at it. i called the type `slasprovenance02` because there's places in the code base which don't like the `.` -- though we could have it diverge from the those if that'd be better

**Review by znewman01 — APPROVED** [review]

**Pull request merged**